docs/docs/concepts/volumes.md

```diff
@@ -0,0 +1,143 @@
+# Volumes
+
+Volumes allow persisting data between runs. When you add a volume,
+`dstack` provisions a network disk in the cloud such as an AWS EBS.
+Then you can mount the volume as a directory in a run and store data there.
+After the run is terminated, the volume can be mounted again and the stored data will persist.
+
+`dstack` supports creating new volumes (a.k.a. `dstack`-managed volumes)
+and also registering existing volumes (a.k.a. external volumes).
+The latter allows accessing data that is already stored on some volume such as pre-processed training data.
+
+!!! info "Backends"
+    Currently, volumes are supported only for `aws`. Support for other backends is coming soon!
+
+!!! info "File system"
+    `dstack` creates an ext4 file system on `dstack`-managed volumes automatically.
+    If you register an external volume, you must ensure it already has a file system.
+
+## Creating new volumes
+
+First create a `volume` configuration file and specify `size` of the volume you'd like to provision:
+
+<div editor-title="new-volume.dstack.yml"> 
+
+```yaml
+type: volume
+name: my-new-volume
+backend: aws
+region: eu-central-1
+size: 100GB
+```
+
+</div>
+
+Then apply the configuration to create the volume:
+
+<div class="termy">
+
+```shell
+$ dstack apply -f new-volume.dstack.yml
+Volume my-new-volume does not exist yet. Create the volume? [y/n]: y
+ NAME           BACKEND  REGION        STATUS     CREATED 
+ my-new-volume  aws      eu-central-1  submitted  now     
+
+```
+
+</div>
+
+The volume is created and can be mounted in runs!
+
+!!! info "Volume parameters"
+    `dstack` has default volume parameters for every backend so you can specify only `size`.
+    On AWS, `dstack` provisions EBS gp3 volumes.
+
+
+## Register existing volumes
+
+If you already have a volume in your cloud account that you'd like to use with `dstack`,
+create a `volume` configuration file with `volume_id` specified:
+
+<div editor-title="external-volume.dstack.yml"> 
+
+```yaml
+type: volume
+name: my-external-volume
+backend: aws
+region: eu-central-1
+volume_id: vol1235
+```
+
+</div>
+
+Then apply the configuration to register the volume:
+
+<div class="termy">
+
+```shell
+$ dstack apply -f external-volume.dstack.yml
+Volume my-external-volume does not exist yet. Create the volume? [y/n]: y
+ NAME                BACKEND  REGION        STATUS     CREATED 
+ my-external-volume  aws      eu-central-1  submitted  now     
+
+```
+
+</div>
+
+The volume is registered and can be mounted in runs!
+
+
+## Mount volumes in runs
+
+Suppose we need to run a dev environment.
+We could mount a volume and store our work there so it's not lost between run restarts or instance interruptions.
+We do it by specifying a list of `volumes`.
+Each item in `volumes` should have `name` of the volume and `path` where the volume should be mounted in the run.
```

Comment:
```suggestion
Each item in `volumes` should have the `name` of the volume and the `path` where the volume should be mounted in the run.
```

Reply:
Stylistically, I wouldn't use articles with field names since they're kind of proper names. Articles ask for additional nouns like "the `name` field", which I'd prefer to avoid to keep it simpler.